components/Login/Login.js

```diff
@@ -0,0 +1,181 @@
+import { useSession, signIn, signOut } from "next-auth/react";
+import LoginIcon from "@/public/icons/LoginIcon.svg";
+import LogoutIcon from "@/public/icons/LogoutIcon.svg";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+import RegularButton from "../Buttons/RegularButton";
+import ButtonWrapper from "../Buttons/ButtonWrapper";
+
+export default function Login({
+  position,
+  changeShowLogOutDialog,
+  showLogOutDialog,
+}) {
+  const { data: session } = useSession();
+
+  const [isSigning, setIsSigning] = useState("");
+
+  function handleLogout() {
+    signOut();
+    changeShowLogOutDialog(false);
+  }
+
+  useEffect(() => {
+    if (isSigning === "in") {
+      handleTimedOutSignIn();
+    }
+
+    function handleTimedOutSignIn() {
+      const signInTimeout = setTimeout(() => {
+        signIn();
+        setIsSigning("");
+      }, 300);
+
+      return () => {
+        clearTimeout(signInTimeout);
+      };
+    }
+  }, [isSigning]);
+
+  useEffect(() => {
+    if (isSigning === "out") {
+      handleTimedOutLogOutDialog();
+    }
+
+    function handleTimedOutLogOutDialog() {
+      const signOutTimeout = setTimeout(() => {
+        changeShowLogOutDialog(true);
+        setIsSigning("");
+      }, 300);
+
+      return () => {
+        clearTimeout(signOutTimeout);
+      };
+    }
+  }, [isSigning, changeShowLogOutDialog]);
+
+  if (session) {
+    return (
+      <>
+        <StyledButton
+          $position={position}
+          onClick={() => {
+            if (position === "header") {
+              changeShowLogOutDialog(true);
+            } else if (position === "menu") {
+              setIsSigning("out");
```

Comment:
So the only difference here would be that if in the menu, the dialog is displayed a bit later. is this due to some animation? you could simplify this by using the timeout directly in this function instead using a useEffect
```jsx
else if (position === "menu") {
      setTimeout(() => {
        changeShowLogOutDialog(true);
      }, 300);

```
then you don't need both useEffects, nor the isSigning state.